Source/WTF/wtf/dragonbox/dragonbox_to_chars.h

```diff
@@ -137,22 +137,22 @@ typedef WTF::double_conversion::StringBuilder StringBuilder;
 template<class Float>
 void ToExponential(Float value, StringBuilder* result_builder)
 {
-    ASSERT((std::is_same<Float, double>::value) || (std::is_same<Float, float>::value));
+    static_assert((std::is_same<Float, double>::value) || (std::is_same<Float, float>::value));
     constexpr size_t buffer_length = 1 + (std::is_same<Float, float>::value ? to_exponential_max_string_length<ieee754_binary32>() : to_exponential_max_string_length<ieee754_binary64>());
     char buffer[buffer_length];
     auto* cursor = detail::to_chars_n<Mode::ToExponential>(value, buffer);
-    result_builder->AddSubstring(buffer, cursor - buffer);
+    result_builder->AddSubstring({ buffer, std::min<size_t>(sizeof(buffer), static_cast<size_t>(cursor - buffer)) });
```

Comment:
Today I learned that a `std::span` may be constructed directly from a ` char buffer[N]` (assuming `buffer` is not a variable-length array or VLA), so this becomes much nicer to read:

```diff
-     result_builder->AddSubstring({ buffer, std::min<size_t>(sizeof(buffer), static_cast<size_t>(cursor - buffer)) });
+     result_builder->AddSubstring(std::span { buffer }.first(cursor - buffer));
```